modules/alloy/src/test/ts/browser/ui/form/HtmlSelectTest.ts

```diff
@@ -1,52 +1,91 @@
+          options: [
+            { value: 'alpha', text: 'Alpha' },
+            { value: 'beta', text: 'Beta' },
+            { value: 'gamma', text: 'Gamma' }
+          ]
+        })
+      )
+    );
+
+    it('TBA: Run tests', () => {
+      const component = store.component();
+
+      simpleAssert('Checking initial value', 'alpha', component);
+      Representing.setValue(component, 'beta');
+      simpleAssert('Checking value after valid set', 'beta', component);
+      Representing.setValue(component, 'delta');
+      simpleAssert('Checking value after invalid set (should still be on beta)', 'beta', component);
+      Representing.setValue(component, 'gamma');
+      simpleAssert('Checking value after valid set (should now be gamma)', 'gamma', component);
+    });
+
+    it('TINY-9679: The options are changed and the selection reset', () => {
+      const component = store.component();
+
+      Assertions.assertStructure(
+        'Checking initial structure',
+        ApproxStructure.build((s, str) => s.element('select', {
+          children: [
+            s.element('option', { value: str.is('alpha'), html: str.is('Alpha') }),
+            s.element('option', { value: str.is('beta'), html: str.is('Beta') }),
+            s.element('option', { value: str.is('gamma'), html: str.is('Gamma') })
+          ]
+        })),
+        component.element
+      );
+
+      // component is reused from previous test, and starts on gamma.
+      simpleAssert('Checking initial value', 'gamma', component);
+      Representing.setValue(component, '');
+      simpleAssert('Checking initial value', 'gamma', component);
+      component.element.dom.value = null;
+      Representing.setValue(component, '');
+      simpleAssert('Checking value after valid set (should now be alpha)', 'alpha', component);
+    });
   });
 
-  const sHasInitialValue = Step.async((next, die) => {
-    GuiSetup.setup((_store, _doc, _body) => GuiFactory.build(
-      HtmlSelect.sketch({
-        dom: { }, // is always a select
-        selectAttributes: {
-          size: 10
-        },
-        selectClasses: [ 'my-test-select' ],
-        data: 'gamma',
-        options: [
-          { value: 'alpha', text: 'Alpha' },
-          { value: 'beta', text: 'Beta' },
-          { value: 'gamma', text: 'Gamma' }
-        ]
-      })
-    ), (_doc, _body, _gui, component, _store) => [
-      Assertions.sAssertStructure(
+  context('Has initial value gamma', () => {
+    const store = GuiSetup.bddSetup(
+      () => GuiFactory.build(
+        HtmlSelect.sketch({
+          dom: {}, // is always a select
+          selectAttributes: {
+            size: 10
+          },
+          selectClasses: [ 'my-test-select' ],
+          data: 'gamma',
+          options: [
+            { value: 'alpha', text: 'Alpha' },
+            { value: 'beta', text: 'Beta' },
+            { value: 'gamma', text: 'Gamma' }
+          ]
+        })
+      )
+    );
+
+    it('TBA: Run tests', () => {
```

Comment:
The label got stuck to the context rather than the specific it-statement, since it needed to be split up to work. Not that it was the most descriptive either.